build/gen_stub.php

```diff
@@ -377,6 +399,78 @@ public function getArgInfoName(): string {
         }
         return 'arginfo_' . $this->name;
     }
+
+    public function getFlags(): string
+    {
+        $flags = sprintf("ZEND_ACC_%s", strtoupper($this->visibility));
+
+        if ($this->isStatic) {
+            $flags .= "|ZEND_ACC_STATIC";
+        }
+
+        if ($this->isFinal) {
+            $flags .= "|ZEND_ACC_FINAL";
+        }
+
+        if ($this->isAbstract) {
+            $flags .= "|ZEND_ACC_ABSTRACT";
+        }
+
+        if ($this->isDeprecated) {
+            $flags .= "|ZEND_ACC_DEPRECATED";
+        }
+
+        return $flags;
+    }
+
+    public function getDeclarationKey(): string
+    {
+        $name = $this->alias ?? $this->name;
+
+        return "$name|$this->cond";
+    }
+
+    public function hasDeclaration(): bool
+    {
+        return $this->isAbstract === false || $this->visibility !== "public";
+    }
+
+    public function getDeclaration(): string
+    {
+        if ($this->hasDeclaration() === false) {
+            return "";
+        }
+
+        if ($this->alias) {
+            $methodAlias = $this->getMethodAlias();
+
+            if ($methodAlias) {
+                return "ZEND_METHOD($this->className, $methodAlias);\n";
```

Comment:
I don't think we can assume that the method alias refers to the same class. I think we have some cases in SPL where implementation is shared across classes. So `getMethodAlias()` should return a class + method pair.

The more principled approach would be to introduce `class FunctionName { public ?string $className; public string $name; }`, which can then be used in both `FuncInfo::$name` and `FuncInfo::$alias` (with the `FuncInfo::$className` field dropped). That should make any code dealing with aliases uniform.